### PR TITLE
HVG-880-embark-passage-tooltip

### PR DESCRIPTION
**Pull request by tarikstafford**

### CODE REVIEW

**Comment by tarikstafford** [comment]
@sampettersson snuck in some fixes for embark plans - removed the hardcoded copy and fixed the button.

**Comment by tarikstafford** [comment]
Screenshot 2021-02-05 at 14 48 29
lmk your thoughts on the spacing as well.

File: Projects/Embark/Sources/EmbarkState.swift [hunk 1]
```diff
@@ -34,6 +34,10 @@ public struct EmbarkState {
     var passageNameSignal: ReadSignal<String?> {
         currentPassageSignal.map { $0?.name }
     }
+    
+    var passageTooltipSignal: ReadSignal<Bool> {
```
Comment (sampettersson) [inline Projects/Embark/Sources/EmbarkState.swift:38]: How about making this return the tooltip instead? Then you could just `if tooltip == nil { hideTabBarButton }`

File: Projects/Embark/Sources/Embark.swift [hunk 2]
```diff
@@ -184,6 +184,33 @@ extension Embark: Presentable {
             }
 
             viewController.navigationItem.leftBarButtonItem = backButton
+            
+            let tooltipButton = UIButton()
+            tooltipButton.isHidden = true
+            tooltipButton.setImage(hCoreUIAssets.infoLarge.image, for: .normal)
+            
+            bag += state.passageTooltipSignal.atOnce().onValue({ (shouldShow) in
+                tooltipButton.isHidden = !shouldShow
```
Comment (sampettersson) [inline Projects/Embark/Sources/Embark.swift:193]: Unsure what is the best practice here, but couldn't we use `        viewController.navigationItem.setRightBarButtonItems([], animated: true)`?

File: Projects/Embark/Sources/Viewables/Tooltip+Alert.swift [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+import hCoreUI
+import Presentation
+import UIKit
+import hGraphQL
+
+typealias Tooltip = GraphQL.EmbarkStoryQuery.Data.EmbarkStory.Passage.Tooltip
+
+extension Tooltip: Presentable {
+    public func materialize() -> (UIViewController, Future<Void>)  {
+
```
Comment (sampettersson) [inline Projects/Embark/Sources/Viewables/Tooltip+Alert.swift:20]: NIT: Don't start with newline

File: Projects/Embark/Sources/Viewables/Tooltip+Alert.swift [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+        }
+        
+        viewController.title = title
+        
+        bag += containerView.didLayoutSignal.onValue({ (_) in
+            viewController.preferredContentSize = containerView.systemLayoutSizeFitting(.zero)
+        })
+        
+        return (viewController, Future { completion in
+
```
Comment (sampettersson) [inline Projects/Embark/Sources/Viewables/Tooltip+Alert.swift:43]: NIT: Don't start with newline

File: Projects/Embark/Sources/Embark.swift [hunk 5]
```diff
@@ -184,6 +184,33 @@ extension Embark: Presentable {
...
+            })
+            
+            let didTapTooltip = tooltipButton.signal(for: .touchUpInside)
+            
+            bag += didTapTooltip
+                .withLatestFrom(state.currentPassageSignal.atOnce().plain())
+                .map { tuple in return tuple.1 }
+                .onValue({ (passage) in
+
```
Comment (sampettersson) [inline Projects/Embark/Sources/Embark.swift:202]: NIT: Don't start with newline

**Review by sampettersson — CHANGES_REQUESTED** [review]

File: Projects/Embark/Sources/Embark.swift [hunk 6]
```diff
@@ -203,7 +230,7 @@ extension Embark: Presentable {
                 )
             )
 
-            viewController.navigationItem.rightBarButtonItem = optionsButton
+            viewController.navigationItem.rightBarButtonItems = [optionsButton, UIBarButtonItem(button: tooltipButton)]
```
Comment (sampettersson) [inline Projects/Embark/Sources/Embark.swift:233]: You could do something like this instead:

```swift
state.currentTooltips.atOnce()
.map { tooltips in tooltips.isEmpty ? [optionsButton] : [optionsButton, tooltipButton] }
.onValue { items in setRightBarButtonItems(items, animated: true) }
```

File: Projects/Embark/Sources/Viewables/Tooltip+Alert.swift (hunk 4 above)
Comment (tarikstafford) [reply Projects/Embark/Sources/Viewables/Tooltip+Alert.swift:43]: I always find it much more readable personally. It's like having a title for a chapter. It makes the title harder to read if the paragraph starts right underneath :)

**Review by sampettersson — APPROVED** [review]

**Pull request merged**